crates/cli/src/config.rs

```diff
@@ -852,6 +852,18 @@ impl Config {
 
         let config = toml::to_string_pretty(&self.home).unwrap();
 
+        // TODO: We currently have a race condition if multiple processes are modifying the config.
+        // If process X and process Y read the config, each make independent changes, and then save
+        // the config, the first writer will have its changes clobbered by the second writer.
+        //
+        // We used to use `Lockfile` to prevent this from happening, but we had other issues with
+        // that approach (see https://github.com/clockworklabs/SpacetimeDB/issues/1339).
+        // We should eventually reintroduce `Lockfile`, with further fixes (see the TODO in
+        // `lockfile.rs`).
```

Comment:
I don't know that we should ever reintroduce `Lockfile`. I think we need a more general solution to the problem. The issue with `Lockfile` is that it may never be released.